### PR TITLE
244 fix remove

### LINKED ISSUE #244: SYNBICT unit tests are very slow
The SYNBICT unit tests take more than 4 times longer with sbol2 than they do with pysbol.

The slowness seems to be centered around removal of objects. SYNBICT removes over 4,000 annotations. On each removal the entire document is searched for the URI being removed. [The block of code](https://github.com/SynBioDex/pySBOL2/blob/4e59b5e411e4dc53083a78bd6ea98c7b0b4d05af/sbol2/property.py#L777-L779) says "Erase nested, hidden TopLevel objects from Document".

First off, the code looks like it is implemented incorrectly. The same [block in libSBOL](https://github.com/SynBioDex/libSBOL/blob/50fb77cec1b043ede08ed315cfac6958e80b9d2f/source/document.h#L1565-L1567) says if the URI is _not_ found, but the sbol2 code says if not None, or _is_ found.

Secondly, what does this code actually do? It looks like if the URI is not found, then the object's doc attribute is set to None. This runs after the object was removed from the Document. Do we need this code?

Removing the code for the SYNBICT test makes them run about 9 times faster than when the code is in place.

### PR DESCRIPTION
**Pull request by bbartley**
Remove inefficient code that was slowing down `remove` operations
Resolve #244

### CODE REVIEW

File: sbol2/property.py [hunk 1]
```diff
@@ -823,10 +820,6 @@ def removeOwnedObject_str(self, uri):
                         # Erase TopLevel objects from Document
                         if self._sbol_owner.getTypeURI() == SBOL_DOCUMENT:
                             del obj.doc.SBOLObjects[rdflib.URIRef(uri)]
-                        # Erase nested, hidden TopLevel objects from Document
-                        if obj.doc is not None and obj.doc.find(uri) is not None:
-                            obj.doc = None  # TODO not sure what this does
-                        self.validate(None)
```
Comment (tcmitchell) [inline sbol2/property.py:829]: I think you deleted one too many lines. I think we want this call to validate. Am I mistaken?

File: test/test_componentdefinition.py [hunk 2]
```diff
@@ -215,6 +215,17 @@ def test_hidden_sequence(self):
                                            rdflib.URIRef(sbol2.SBOL_SEQUENCE_PROPERTY),
                                            rdflib.URIRef(seq.identity)))
 
+    @unittest.expectedFailure
```
Comment (tcmitchell) [inline test/test_componentdefinition.py:218]: Is this the test that demonstrates #272 ? If so, can a comment be added to point to issue 272?

File: sbol2/property.py (hunk 1 above)
Comment (bbartley) [reply sbol2/property.py:829]: Yep that was an unintentional merge foul

File: test/test_componentdefinition.py (hunk 2 above)
Comment (bbartley) [reply test/test_componentdefinition.py]: Done

**Pull request merged**